### DISCUSSION TITLE
Does Serverless support ES modules?

### DISCUSSION THREAD

**Discussion by nthouliss** [serverless/serverless#11102] [Q&A]
I'm trying out Serverless for the first time and am trying to deploy my existing lambdas (currently deployed with Terraform and written in vanilla javascript). They are of `type: "module"` but when deploying with serverless it looks like the file being generated by Serverless uses `require` rather than `import` and is causing errors. 

```
        "Error [ERR_REQUIRE_ESM] [ERR_REQUIRE_ESM]: Must use import to load ES Module: /var/task/selector/src/index.js",
        "require() of ES modules is not supported.",
        "require() of /var/task/selector/src/index.js from /var/task/s_selector.js is an ES module file as it is a .js file whose nearest parent package.json contains \"type\": \"module\" which defines all .js files in that package scope as ES modules.",
        "Instead rename index.js to end in .cjs, change the requiring code to use import(), or remove \"type\": \"module\" from /var/task/selector/src/package.json.",
        "",
        "    at new NodeError (internal/errors.js:322:7)",
        "    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)",
        "    at Module.load (internal/modules/cjs/loader.js:950:32)",
        "    at Function.Module._load (internal/modules/cjs/loader.js:790:12)",
        "    at Module.require (internal/modules/cjs/loader.js:974:19)",
        "    at Module._require.i.require (/var/task/serverless_sdk/index.js:9:73397)",
        "    at Module._require.i.require (/var/task/serverless_sdk/index.js:9:73397)",
        "    at Module._require.i.require (/var/task/serverless_sdk/index.js:9:73397)",
        "    at require (internal/modules/cjs/helpers.js:101:18)",
        "    at Object.<anonymous> (/var/task/s_selector.js:25:23)"
```

Is there a way to get this generated file to use import instead? I've already tried to use `serverless-esbuild` but there isn't support for top-level awaits.

**Comment by pgrzesik** [comment]
Hey @nthouliss - thanks a lot for reporting and sorry you've run into troubles. Just to confirm - you're using integration with Serverless Dashboard, correct?

**Reply by nthouliss** [reply]
Yes, I'm using integration with Serverless Dashboard.

**Reply by nthouliss** [reply]
Should I raise this as a separate issue @pgrzesik? Or is this not something I'm going to be able to do with Serverless?

**Reply by pgrzesik** [reply]
Hey @nthouliss - sorry for not responding earlier - I've reported the issue internally last week and did some digging - after we evaluate if we can add support for ES Modules in a reasonable time I'll let you know. As it stands right now, current wrapping logic supports Common JS only.

**Reply by nthouliss** [reply]
No worries, thanks for the update 😄
